### PR TITLE
Fix coercion from ObjectID to String

### PR DESCRIPTION
**Pull request by raymondfeng**
/to @ritch 

This patch is result of https://groups.google.com/forum/#!topic/loopbackjs/zUQY1aQ5oGA

### CODE REVIEW

**Comment by pulkitsinghal** [comment]
@raymondfeng - wanted to let you know that it worked

I added `"loopback-datasource-juggler": "git://github.com/strongloop/loopback-datasource-juggler.git#eb6fa41"` to `package.json` and nuked the db to start over and my boot script finished without any issues.

File: lib/dao.js [hunk 1]
```diff
@@ -1135,12 +1135,7 @@ DataAccessObject._coerce = function (where) {
     }
     // Check there is an operator
     var operator = null;
-    if ('object' === typeof val) {
-      if (Object.keys(val).length !== 1) {
-        // Skip if there are not only one properties
-        // as the assumption for operators is not true here
-        continue;
-      }
+    if (val.constructor === Object) {
```
Comment (ritch) [inline lib/dao.js]: This seems a bit more strict than we need... Isn't `typeof val === 'object' && !Array.isArray(val)` what we are looking for?

**Comment by ritch** [comment]
LGTM

File: lib/dao.js (hunk 1 above)
Comment (raymondfeng) [inline lib/dao.js]: One case that we want to skip if mongodb.ObjectID.

**Pull request merged**